File: pagamento-oneclick/su-pagina-di-cassa/php/pagamento.php
                                            
<?php

//  Pagamento OneClik - Su pagina di cassa - Avvio pagamento

// Alias e chiave segreta 
$ALIAS = '<ALIAS>'; // Sostituire con il valore fornito da Nexi
$CHIAVESEGRETA = '<CHIAVE SEGRETA PER CALCOLO MAC>'; // Sostituire con il valore fornito da Nexi

$requestUrl = "https://int-ecommerce.nexi.it/ecomm/ecomm/DispatcherServlet";
$merchantServerUrl = "https://" . $_SERVER['HTTP_HOST'] . "/xpay/php/pagamento_semplice/one_click/";

$codTrans = "TESTPS_" . date('YmdHis');
$divisa = "EUR";
$importo = 5000;

$numContratto = "NC_TEST_" . date('YmdHis');
$gruppo = '<GRUPPO>'; // Sostituire con il valore fornito da Nexi

// Calcolo MAC
$mac = sha1('codTrans=' . $codTrans . 'divisa=' . $divisa . 'importo=' . $importo . 'gruppo=' . $gruppo . 'num_contratto=' . $numContratto . $CHIAVESEGRETA);

// Parametri obbligatori
$obbligatori = array(
    'alias' => $ALIAS,
    'importo' => $importo,
    'divisa' => $divisa,
    'codTrans' => $codTrans,
    'url' => $merchantServerUrl . "esito.php",
    'url_back' => $merchantServerUrl . "annullo.php",
    'mac' => $mac,
    'num_contratto' => $numContratto,
    'tipo_servizio' => 'paga_1click',
   
    );

// Parametri facoltativi
$facoltativi = array(
);

$requestParams = array_merge($obbligatori, $facoltativi);

?>

<html>
    <head></head>
    <body>
        <form method='POST' action='<?php echo $requestUrl ?>'>
            <?php foreach ($requestParams as $name => $value) { ?>
                <input type='hidden' name='<?php echo $name; ?>' value='<?php echo htmlentities($value); ?>' />
            <?php } ?>
            
            <input type='submit' value='VAI ALLA PAGINA DI CASSA' />
        </form>
    </body>
</html>
                    
                

File: pagamento-oneclick/su-pagina-di-cassa/dotnet/pagamento.cs
                                            
//  Pagamento OneClik - Su pagina di cassa - Avvio pagamento

using System;
using System.Collections.Generic;
using System.Text;
using System.Security.Cryptography;

namespace XPay
{
    class SampleXPayRequest
    {
        static void Main(string[] args)
        {

            // Alias e chiave segreta
            string ALIAS = "<ALIAS>"; // Sostituire con il valore fornito da Nexi
            string CHIAVESEGRETA = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Sostituire con il valore fornito da Nexi

            string requestUrl = "https://int-ecommerce.nexi.it/ecomm/ecomm/DispatcherServlet";
            string merchantServerUrl = "https://my-server.example.tdl/";

            DateTime data = DateTime.Now;
            string codTrans = "TESTPS_" + data.ToString("yyyyMMddHHmmss");
            string divisa = "EUR";
            string importo = "5000";
            
            string numContratto = "NC_TEST_" + data.ToString("yyyyMMddHHmmss");
            string gruppo = "<GRUPPO>" // Sostituire con il valore fornito da Nexi

            // Calcolo MAC
            SampleXPayRequest sampleXPay = new SampleXPayRequest();
            string mac = sampleXPay.HashMac("codTrans=" + codTrans + "divisa=" + divisa + "importo=" + importo + "gruppo=" + gruppo + "num_contratto=" + numContratto + CHIAVESEGRETA);

            var requestParams = new Dictionary<string, string>();
            // Parametri obbligatori
            requestParams["alias"] = ALIAS;
            requestParams["importo"] = importo;
            requestParams["divisa"] = divisa;
            requestParams["codTrans"] = codTrans;
            requestParams["url"] = merchantServerUrl + "esito.aspx";
            requestParams["url_back"] = merchantServerUrl + "annullo.aspx";
            requestParams["mac"] = mac;
            requestParams["num_contratto"] = numContratto;
            requestParams["tipo_servizio"] = "paga_1click";
            

            // Parametri facoltativi
			
            
            /**
            * Creare un form html con metodo post verso requestUrl con campi hidden contenenti requestParams
            */
        }

        public string HashMac(string s)
        {
            byte[] bytes = Encoding.UTF8.GetBytes(s);

            var sha1 = SHA1.Create();
            byte[] hashBytes = sha1.ComputeHash(bytes);

            return HexStringFromBytes(hashBytes);
        }

        public static string HexStringFromBytes(byte[] bytes)
        {
            var sb = new StringBuilder();
            foreach (byte b in bytes)
            {
                var hex = b.ToString("x2");
                sb.Append(hex);
            }
            return sb.ToString();
        }

    }
}
                    
                